api/models/trial.js

```diff
@@ -210,6 +221,17 @@ const Trial = BaseModel.extend({
 
       return discrepancies;
     },
+    is_registered() {
+      // Be aware that this depends on `source_id` to be accurate
+      // So if a trial has a `nct` source but a `pubmed` source_id,
+      // then the problem is with the processor
+      let status;
+      const source = this.sources[this.attributes.source_id] || { type: null };
+      if (source.type !== null) {
+        status = source.type === 'register';
+      }
+      return status;
```

Comment:
Good points, except that node would complain when no source is returned: trying to access `source.type` when `source` is `undefined` (that's why I place a stub `source` having only `type` if none found. I cannot reproduce this error in the node console though, so I guess it has something to do with one of the many libraries in a request's life cycle.

I'm doing the others now, but I don't guarantee I will be successful removing the `{ type: null }` object from there. I'll try again though, see what happens (maybe the error was somewhere else).